src/Search/ProductFinder.php

```diff
@@ -183,4 +185,33 @@ public function getResults(): Collection
         }
         return $this->queryBuilder->get();
     }
+
+    /** @see Builder::simplePaginate() */
+    public function simplePaginate($perPage = 15, $columns = ['*'], $pageName = 'page', $page = null): Paginator
+    {
+        if ($this->excludeInactiveProducts) {
+            $this->queryBuilder->whereIn(
+                'state',
+                ProductStateProxy::getActiveStates()
+            );
+        }
+        return $this->queryBuilder->simplePaginate($perPage, $columns, $pageName, $page);
+    }
+
+    /** @see Builder::paginate() */
+    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page', $page = null): LengthAwarePaginator
+    {
+        if ($this->excludeInactiveProducts) {
```

Comment:
@fulopattila122 is there a specific reason to use `$excludeInactiveProducts` as member variable to check before every get operation, instead of just applying the required query part in the `withInactiveProducts()` once? If not I can change the function and remove other unnecessary parts.

Reply:
I like the `withInactiveProducts()` approach, please add it to this PR. Thanks!

Reply:
I feel dumb. After taking a look at the code to implement it, it looks like it can't be done that way.
Only thing that I can think of that eliminates checking before taking actions is making that change in constructor. But I think that will be more complex since it requires finding and removing the condition when calling `withInactiveProducts()`.
So I took a different route.

Reply:
Then realized that using global scopes are better, so I reverted that change and I think it is now in its final form.